services/sysinfo/server/sysinfo_linux.go

```diff
@@ -0,0 +1,24 @@
+//go:build linux
+// +build linux
+
+/*
+Copyright (c) 2023 Snowflake Inc. All rights reserved.
+
+	Licensed under the Apache License, Version 2.0 (the
+	"License"); you may not use this file except in compliance
+	with the License.  You may obtain a copy of the License at
+
+	  http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing,
+	software distributed under the License is distributed on an
+	"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+	KIND, either express or implied.  See the License for the
+	specific language governing permissions and limitations
+	under the License.
+*/
+package server
+
+var getUptimeFilePath = func() (string, error) {
+	return "/proc/uptime", nil
```

Comment:
Maybe we could use a ready package vs reading it directly. We could use https://pkg.go.dev/golang.org/x/sys/unix.
And then something like
```
var info unix.Sysinfo_t
if err := unix.Sysinfo(&info); err != nil {
    fmt.Println("got error:", err)
}
fmt.Println(time.Duration(info.Uptime) * time.Second)
```

Reply:
+1

The way to get uptime might vary a lot across operating systems, so implementing a `func() (time.Duration, error)` in both _linux and _default seems better than implementing a `func() (string, error)`.

Reply:
You’re right, in linux we can get the uptime by reading a specific file, but things may not work for other os when we need to support in the future. So providing the `getUptime() `(time.Duration, error) can be more robust. And in linux, both solutions (reading from `/proc/uptime` and` unix.Sysinfo()`) can populate the uptime info well, but the latter solution makes code more concise since it provides a higher-level interface to get system information. So let me change the solution.